### ISSUE TITLE
Invalid pass number (1) for Graphics.Blit when reloading scene from asset bundle

### ISSUE THREAD

**Issue by magnopus-bueno** [Unity-Technologies/PostProcessing#793]
We're currently doing some art look development in a project that makes heavy use of asset bundles. What we're noticing is that the first time we load a bundle and a scene, everything works as expected. However, upon unloading the bundle, reloading it, and loading the same scene again, post-processing appears to break. 

If "Stop NaN Propagation" is unchecked, the entire scene will be broken-material pink. If it's checked, the console will spam "Invalid pass number (1) for Graphics.Blit".

We are currently using Post-processing stack v2 with Unity version 2018.3.2f1 on Mac, building for the iOS platform. We notice the problem both in editor and build.

Here is an example project demonstrating the problem: https://drive.google.com/file/d/18RKuwyTxLNdgNIXJGWYjKAB2ovSEmvW1/view?usp=sharing

Open up and play the Bootstrap scene. Once the BugExample scene has loaded, click the button and the problem will present itself.

**Comment by tayfundoker** [comment]
Same but mine is only in webgl, no problem in pc build

**Comment by jshmrsn** [comment]
We are also getting this issue. The issue is due to the fact that static variables in RuntimeUtilities holds onto shaders from the PostProcessResources. Those shaders can be deleted - in our case because we load PostProcessResources from an asset bundle, and those asset bundles can be unloaded and re-loaded within a game session.

As the code is structured now, those static variables need to be set to null whenever we unload asset bundles, so that they will later be re-initialized with new shaders when asset bundles are loaded again.

We created a static function on RuntimeUtilities to set all of its resource static to null, and we call that function in PostProcessLayer.SetupContext, right after RuntimeUtilities.s_Resources is assigned.

RuntimeUtilities.cs
`public static void ResetResources() {
            s_CopyStdMaterial = null;
            s_CopySheet = null;
            s_CopyMaterial = null;
            s_CopyFromTexArrayMaterial = null;
            s_CopyStdFromDoubleWideMaterial = null;
            
            s_CopySheet = null;
            s_CopyFromTexArraySheet = null;
        }`

PostProcessLayer.cs
`        void SetupContext(PostProcessRenderContext context)
        {
            RuntimeUtilities.s_Resources = m_Resources;
            RuntimeUtilities.ResetResources();
`

**Comment by wtowns** [comment]
@jshmrsn , you're a modern hero: your workaround works perfectly.  How did you figure out that this was the problem?

**Comment by jshmrsn** [comment]
@wtowns Glad the workaround is working for you as well. I found it by adding a serialized field to PostProcessLayer just for the purpose of viewing in the Inspector in debug mode (since it has a custom inspector editor class). The field was a list of materials, which I then shoved every material the PostProcessLayer ever tried to use into, so that I could view all of those materials in the Inspector.

After reproducing the bug in the editor, I could see that most materials were fine, except for the few materials that were created in the static fields, which had missing/deleted references for their shaders. The other materials are recreated in each new instance of PostProcessLayer, but the statics hang around forever so it kinda made sense what was going on there.

By the way, I heard from a coworker that you and I previously worked in the same studio, though I left a couple years before you joined :)
